### PR TITLE
Clique: Force priority location for "final location" and other minor tweaks.

### PR DESCRIPTION
**Pull request by ThePhar**
## What is this fixing or adding?
Forces "The Button" to be a priority location.

Also tweaked names of items and small change to rules to generate a better spoiler play-through... for the 1 person who cares about that.

## How was this tested?
Generated a game in hard mode and a game not in hard mode.

## If this makes graphical changes, please attach screenshots.
🟥
👆

### CODE REVIEW

**Comment by Berserker66** [comment]
image

**Comment by ThePhar** [comment]
For some reason it's acting as if it's a new world definition instead of taking the small tweaks I made. I'll look into it later today.

**Comment by Alchav** [comment]
> For some reason it's acting as if it's a new world definition instead of taking the small tweaks I made. I'll look into it later today.

I think that's just because you didn't update with main so it thinks it has a conflict with the earlier PR to add Clique

**Comment by ThePhar** [comment]
Alright, should be all resolved now.

**Comment by ThePhar** [comment]
Ran beta test, was good.

File: worlds/clique/__init__.py [hunk 1]
```diff
@@ -32,78 +22,69 @@ class CliqueWorld(World):
...
+        self.multiworld.completion_condition[self.player] = lambda state: \
+            state.has("Feeling of Satisfaction", self.player)
 
 
 def create_region(world: MultiWorld, player: int, name: str, locations=None, exits=None):
     region = Region(name, player, world)
     if locations:
         for location_name in locations.keys():
-            location = CliqueLocation(player, location_name, locations[location_name], region)
+            location = Location(player, location_name, locations[location_name], region)
```
[the remark this answers is not in the document]
Comment (ThePhar) [reply worlds/clique/__init__.py:83]: Was just trying to be more lean. Fair enough, will make a tweak.

**Comment by ThePhar** [comment]
Let's give that a whirl.

**Comment by ThePhar** [comment]
As Berserker painfully pointed out, this really needs to wait on #1659 to be merged first. Going to do the tweaks to that one tomorrow when I wake up.

**Comment by ThePhar** [comment]
Should probably also mention this will also be a fix for #1460

File: worlds/clique/__init__.py [hunk 2]
```diff
@@ -32,78 +22,70 @@ class CliqueWorld(World):
...
-        return getattr(self.multiworld, name)[self.player]
+    item_name_to_id = {
+        "Feeling of Satisfaction": 69696969,
+        "Button Activation":       69696968,
+    }
 
-    def fill_slot_data(self) -> dict:
-        return {option_name: self.get_setting(option_name).value for option_name in self.option_definitions}
+    def create_item(self, name: str) -> Item:
+        return Item(name, ItemClassification.progression, self.item_name_to_id[name], self.player)
```
Comment (Berserker66) [inline worlds/clique/__init__.py:41]: Do keep in mind that .game is not set like this, which makes it harder to figure out which game an item came from in debugging.

Comment (ThePhar) [reply worlds/clique/__init__.py:41]: Very fair. Was hoping to just remove the extra definitions, but probably best I don't. Will correct.

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**